calicoctl/commands/node/status.go

```diff
@@ -129,20 +131,40 @@ func printBGPPeers(ipv string) {
 		os.Exit(1)
 	}
 
-	buf := make([]byte, 1024)
+	// BIRD socket read timeout.
+	timeOut := 2 * time.Second
+	bufReader := bufio.NewReader(c)
 
-	n, err := c.Read(buf[:])
-	if err != nil {
-		fmt.Printf("Error: %v\n", err)
+	birdOut := ""
+
+	for {
+		// Set a time-out for reading from the socket connection.
+		// Read operation will fail if no data is received until the timeout.
+		c.SetReadDeadline(time.Now().Add(timeOut))
```

Comment:
Do we need to keep setting the timeout?  Surely we can do this once outside the for loop.